homeassistant/components/shelly/coordinator.py

```diff
@@ -619,6 +619,8 @@ def _async_handle_update(
             self.hass.async_create_task(self._async_disconnected())
         elif update_type is RpcUpdateType.STATUS:
             self.async_set_updated_data(None)
+            if self.sleep_period:
+                device_update_info(self.hass, self.device, self.entry)
```

Comment:
I think it would be better to compare if the version changed before we try to update it, since thins update happens on every status from the device, what I noticed in Gen1 that calling this method does many comparisons internally.

Example in Gen1:
https://github.com/home-assistant/core/blob/aa75ecd7570dce4eb6c16924f84319a16c398f7d/homeassistant/components/shelly/coordinator.py#L381-L382

Reply:
Good point!